processor/otel/consumer.go

```diff
@@ -397,27 +399,40 @@ func parseSpan(span *tracepb.Span, event *model.Span) {
 				db.Statement = &v.StringValue.Value
 				isDBSpan = true
 			case "db.instance":
-				db.Instance = &v.StringValue.Value
+				val := truncate(v.StringValue.Value)
+				db.Instance = &val
 				isDBSpan = true
 			case "db.type":
-				db.Type = &v.StringValue.Value
+				val := truncate(v.StringValue.Value)
+				db.Type = &val
 				isDBSpan = true
 			case "db.user":
-				db.UserName = &v.StringValue.Value
+				val := truncate(v.StringValue.Value)
+				db.UserName = &val
 				isDBSpan = true
 			case "peer.address":
 				val := truncate(v.StringValue.Value)
-				destination.Address = &val
 				destinationService.Resource = &val
-			case "peer.hostname":
-				if destination.Address == nil {
-					val := truncate(v.StringValue.Value)
+				if !strings.ContainsRune(val, ':') || net.ParseIP(val) != nil {
+					// peer.address is not necessarily a hostname
+					// or IP address; it could be something like
+					// a JDBC connection string or ip:port. Ignore
+					// values containing colons, except for IPv6.
 					destination.Address = &val
 				}
+			case "peer.hostname", "peer.ipv4", "peer.ipv6":
+				val := truncate(v.StringValue.Value)
+				destination.Address = &val
 			case "peer.service":
-				destinationService.Name = &v.StringValue.Value
+				val := truncate(v.StringValue.Value)
+				destinationService.Name = &val
+				if destinationService.Resource == nil {
+					// Prefer using peer.address for resource.
+					destinationService.Resource = &v.StringValue.Value
```

Comment:
Why are you not also using the truncated value here?

Reply:
Accidental, thanks for picking up - will fix